subsys/bluetooth/audio/cap_initiator.c

```diff
@@ -868,6 +869,66 @@ void bt_cap_initiator_enabled(struct bt_cap_stream *cap_stream)
 	}
 }
 
+void bt_cap_initiator_almost_started(struct bt_cap_stream *cap_stream)
+{
+	int err;
+	struct bt_bap_ep *ep = cap_stream->bap_stream.ep;
+	struct bt_bap_ep *paired_ep = bt_bap_iso_get_paired_ep(ep);
+	struct bt_cap_common_proc *active_proc = bt_cap_common_get_active_proc();
+	struct bt_cap_stream *next_cap_stream = NULL;
+
+	if (!bt_cap_common_stream_in_active_proc(cap_stream)) {
+		/* State change happened outside of a procedure; ignore */
+		return;
+	}
+
+	if (ep->dir == BT_AUDIO_DIR_SINK && paired_ep != NULL) {
+		/* If there are 2 ASEs in the CIS, continue with the one directed to Source ASE,
+		 * in case there will be Write Long used at Receiver Start Ready operation.
+		 */
+
+		if (paired_ep->receiver_ready == false) {
+			/* If the Source ep was placed after the Sink ep in the bt_cap_unicast_audio_start_param,
+			 * let's start the Source ep here.
+			 */
+			bt_bap_stream_start(paired_ep->stream);
+		}
+
+		return;
+	}
+
+	for (size_t i = 0; i < active_proc->proc_cnt; i++) {
+		struct bt_cap_stream *stream = active_proc->proc_param.initiator[i].stream;
+		struct bt_iso_chan *iso_chan = bt_bap_stream_iso_chan_get(&stream->bap_stream);
+
+		if (iso_chan != NULL && iso_chan->state == BT_ISO_STATE_DISCONNECTED) {
+			next_cap_stream = stream;
+			break;
+		}
+	}
+
+	if (next_cap_stream == NULL) {
+		/* The CISes are set up and Receiver Start Ready operations have
+		 * been sent. All that's left is to wait for Streaming notifications
+		 * if they if they haven't arrived yet.
+		 */
+		return;
+	}
+
+	/* Not yet finished, start next */
+	err = bt_bap_stream_start(&next_cap_stream->bap_stream);
```

Comment:
This won't work when starting multiple source streams on a single device: If you trigger the `bt_bap_stream_start` on the CIS connect, rather than the state change, then that means that we will have the previous `receiver start ready` still unhandled. 

It would be a non-issue if we had https://github.com/zephyrproject-rtos/zephyr/issues/55458 implemented, where we would start all streams in a single ISO/GATT request.